### PR TITLE
Convert tests to pytest

### PR DESCRIPTION
**Pull request by cjacoby**
Adresses #43.

I tried to keep the test conversion simple, such that I use the `py.test` paradigm as much as possible, while keeping the intent of the original test.

### CODE REVIEW

**Comment by bmcfee** [comment]
Thanks for taking this one! I'll look at it tomorrow.

File: tests/test_buffered_streams.py [hunk 1]
```diff
@@ -1,75 +1,73 @@
...
 
+@pytest.mark.parametrize('n1', [5, 10, 15])
+@pytest.mark.parametrize('n2', [5, 10, 15])
+def test_batch_length__not_equal(n1, n2):
+    if n1 != n2:
+        generator, n = T.__zip_generator(3, n1, n2), n1
+
+        with pytest.raises(pescador.PescadorError):
+            for batch in generator:
+                assert pescador.buffered.batch_length(batch) == n
```
Comment (bmcfee) [inline tests/test_buffered_streams.py]: This test might be simpler if you tuple-ize (n1, n2) as parameters, and then mark out a couple of specific cases `(5, 10)` as failures.  [Here](https://github.com/bmcfee/pumpp/blob/master/tests/test_task.py#L419-L423) is an example illustrating how to do this.

**Comment by bmcfee** [comment]
Hot diggity -- this got coverage working in the zmq parallel process?

**Comment by bmcfee** [comment]
Also, I cleared the dependency cache and restarted the build.  Hopefully that will clear up the push failure.

**Comment by bmcfee** [comment]
@cjacoby is there anything else you want to add here, or is it good to go?

**Comment by cjacoby** [comment]
I think it's good to go. Yay for py.test!

**Pull request merged**